modules/api/src/main/java/org/apache/ignite/table/criteria/CriteriaQuerySource.java

```diff
@@ -38,19 +38,33 @@ public interface CriteriaQuerySource<T> {
      * @throws CriteriaException If failed.
      */
     default Cursor<T> query(@Nullable Transaction tx, @Nullable Criteria criteria) {
-        return query(tx, criteria, null);
+        return query(tx, criteria, null, null);
+    }
+
+    /**
+     * Executes predicate-based criteria query.
+     *
+     * @param tx Transaction to execute the query within or {@code null} to run within implicit transaction.
+     * @param criteria The predicate to filter entries or {@code null} to return all entries from the underlying table.
+     * @param indexName The name of the index to force usage of this index in the query.
+     * @return Iterator with query results.
+     * @throws CriteriaException If failed.
+     */
+    default Cursor<T> query(@Nullable Transaction tx, @Nullable Criteria criteria, @Nullable String indexName) {
+        return query(tx, criteria, indexName, null);
     }
 
     /**
      * Executes a predicate-based criteria query.
      *
      * @param tx Transaction to execute the query within or {@code null} to run within implicit transaction.
      * @param criteria The predicate to filter entries or {@code null} to return all entries from the underlying table.
+     * @param indexName The name of the index to force usage of this index in the query.
      * @param opts Criteria query options or {@code null} to use default.
      * @return Iterator with query results.
      * @throws CriteriaException If failed.
      */
-    Cursor<T> query(@Nullable Transaction tx, @Nullable Criteria criteria, @Nullable CriteriaQueryOptions opts);
+    Cursor<T> query(@Nullable Transaction tx, @Nullable Criteria criteria, @Nullable String indexName, @Nullable CriteriaQueryOptions opts);
```

Comment:
All method parameters are optional.
Does the next call make sense?
"query(null, null, null, null)" ?

Reply:
Yes, this query will be run within implicit transaction and return all entries from the underlying table.